### DISCUSSION TITLE
Integrating large dataset

### DISCUSSION THREAD

**Discussion by deconvolute** [satijalab/seurat#7417] [Q&A]
HI,

I am trying to integrate a dataset with 300,000 cells but I encounter the following error:
Integrating data
Merging dataset 18 41 32 16 40 9 44 8 22 12 4 11 28 25 46 20 14 37 42 62 35 into 17 61 43 52 33 5 58 24 19 48 6 13 27 26 30 29 51 50 49 54 36 59 47 57 7 56
Extracting anchors for merged samples
Finding integration vectors
Error in .T2C(newTMat(i = c(ij1[, 1], ij2[, 1]), j = c(ij1[, 2], ij2[,  : 
  unable to coerce from TsparseMatrix to [CR]sparseMatrixwhen length of 'i' slot exceeds 2^31-1
Calls: IntegrateData ... FindIntegrationMatrix -> - -> - -> .Arith.Csparse -> .T2C
Execution halted

**Comment by BenjaminDEMAILLE** [comment]
Same

**Comment by anemartinezlarrinaga2898** [comment]
Did anyone solved this?

**Comment by MikedeKokkie** [comment]
Having the same error. Bump?

**Comment by sc01619** [comment]
Having the same issue bump

**Comment by jvelghe** [comment]
Also having this error and not sure why yet... trying to integrate just under 300,000 cells as well.

**Comment by ganpha** [comment]
I'm trying to integrate >300,000 cells and get the same error. How to resolve this issue. Thanks!

**Comment by iddryg** [comment]
I had the same issue. As far as I know there isn't a true solution yet, but there are some alternatives shown here: [https://satijalab.org/seurat/archive/v3.0/integration.html](https://satijalab.org/seurat/archive/v3.0/integration.html)

Mainly options 3 and 4:
- specify a reference in your FindIntegrationAnchors() call (choose one of your datasets, you can use reference=1 to pick the first of the datasets). It will build the anchors around that reference dataset instead of making new anchors every time. 
- specify reduction="rpca" in your FindIntegrationAnchors() call. This uses reciprocal PCA instead of CCA.
